### ISSUE TITLE
copy-to usage

### ISSUE THREAD

**Issue by ural-mtb** [dita-ot/dita-ot#2989]
## Expected Behavior
Hello everyone! I have faced the situation when the theory is not enough to understand how a certain feature works. I am talking about the **copy-to** attribute usage. I know that it is quite a basic thing, but I cannot find any samples on the Internet. 

The description at [oxygenxml.com](https://www.oxygenxml.com/dita/styleguide/webhelp-feedback/Artefact/Maps/c_copy-to_Attribute.html) says: 
"_You can use the copy-to attribute of the ditamap topicref element when you need the output file name to differ from what would otherwise be automatically generated, or **when you need two or more identical or almost identical copies of the same topic in the output**. The copies of the original topic can have different generated link text and generated short descriptions._"

## Actual Behavior
I tried to add this attribute, but I cannot figure out the following:
- Where should the copy-to attribute go? After the original file that I want to duplicate?
- How I can actually reuse the duplicated topic? So if I want to use `topic_1.dita` one more time in the same map, how should I do it? 
- I tried several options and could not find the right place for **topicmeta** usage - to add an alternative title, for example.

## Environment
* DITA-OT version: 3.0.4
* Operating system and version: Windows
* How did you run DITA-OT: The `dita` command
* Transformation type: PDF

Thank you in advance! 

[new_map.zip](https://github.com/dita-ot/dita-ot/files/2066270/new_map.zip)

**Comment by drmacro** [comment]
Here is the description of @copy-to in the DITA spec:


@copy-to
    Use the @copy-to attribute on the <topicref> element to provide a different resource name for a particular instance of a resource referenced by the <topicref> (for example, to separate out the different versions of the topic, rather than combining them on output). If applicable, the @copy-to value can include path information. The links and navigation associated with that instance will point to a copy of the topic with the file name you specified. Applications MAY support @copy-to for references to local non-DITA resources. The @copy-to attribute is not supported for references to resources where the effective value for @scope is "peer" or "external".

    Use the <linktext> and <shortdesc> in the <topicref>'s <topicmeta> to provide a unique name and short description for the new copy.

(http://docs.oasis-open.org/dita/dita/v1.3/errata01/os/complete/part1-base/langRef/attributes/topicrefElementAttributes.html#topicref-element-atts)

The Oxygen language you quote is not quite accurate in that @copy-to does not specify the effective *result* filename, it specifies the effective *source* filename, which may (or may not) then affect the result filename if the referenced topic happens to result in a separate file in the rendition (e.g., you're generating HTML).

The best way to refer to topics on which you've used @copy-to is to use keys and key references. That ensures that you're referring to the exact use of the topic you intend, e.g.:

```
<topicref keys="topic-a-use-1" href="topics/topic-a.dita"/>
<topicref keys="topic-a-use-2" href="topics/topic-a.dita" copy-to="topics/topic-a-02.dita"/>
```

Then in a topic an xref like `<xref keyref="topic-a-use-2"/>` can only mean "use the use of topic-a.dita with the key 'topic-a-use-2', which happens to have the effective source file name 'topics/topic-a-02.dita'"

If for some reason the use of keys is not possible, then you can refer to the effective URL of the topic as defined by @copy-to and the processor should correctly resolve it. That is, the behavior should be identical to having made a literal copy of the topic with the name defined by @copy-to and referenced by that name.

**Comment by ural-mtb** [comment]
Thank you for the feedback, @drmacro!

I haven't used keys yet, so it is kinda tricky at the moment. 

What will be the best practice if I need to use the same topic twice in the same output? The first time it must be the original file without any changes, while the second time the same topic should have another title, for example. It can be an intro for two different chapters, for example. The content should remain the same, I just want to use different titles.

**Comment by raducoravu** [comment]
@drmacro That particular topic from our web site is part of the DITA Style Guide by dr. Tony Self. Being a style guide it is less technical and probably tries to focus on the general benefits of using copy-to instead of copying the description in the specs.

**Comment by ural-mtb** [comment]
Here is what I mean by actual usage. Let's say we have the following map (check the _test_map.ditamap_) and we want to use `topic_2.dita` for the second time between `topic_5.dita` and `topic_6.dita`. And for some reason we want to change its title (when it is used for the second time).

In this case: should I use the `copy-to` attribute to `topic_2.dita` when it is used for the second time? And to add `topicmeta` element with `navtitle` (check the _test_map_2.ditamap_)? 

P.S. I know it is basic stuff and it's probably boring for you guys, but I am really confused. Thanks! 

[test_map.zip](https://github.com/dita-ot/dita-ot/files/2067126/test_map.zip)

**Comment by drmacro** [comment]
Note that these types of DITA usage questions are best asked on the Yahoo DITA Users group since these are not really DITA Open Toolkit issue.

Short answer to your question: you can override the navigation title in the topicref to a topic but you cannot change the title. To get different titles over the same body content there are different possible techniques that are too involved to discuss here.

**Comment by ural-mtb** [comment]
Thanks for the comment, @drmacro! I will definitely check the Yahoo DITA Users group. 

Meanwhile, I've discussed the issue with some people and tried to figure out how often they use the `copy-to` feature. I believe that this option is qute rare, after all. And (it's an easy way to avoid finding the direct solution, though) I guess it is always possible to create a separate topic and use a conref there, which can contain the general shared information. At least it looks like a simple way out. 

Samples of code that I found on pages of Oxygen help and DITA specification do not provide the result that I've expected to get, unfortunately (I even copied the code, but still no result).

**Comment by drmacro** [comment]
If you feel like the OT is not behaving as you would expect then you should create a small test case that demonstrates the behavior and include it with the issue. There have definitely been bugs related to @copy-to processing in various releases so it's quite possible that you're simply encountering a bug.

**Comment by ural-mtb** [comment]
https://www.oxygenxml.com/dita/styleguide/webhelp-feedback/Artefact/Maps/c_copy-to_Attribute.html
... For example, you may want the information in a topic about turbochargers to appear in the output twice, and to be referenced in the Related Topics link group in one instance as "XT Turbocharger Explained", but in another output topic's Related Topics link group as "WRX Turbocharger Explained". In this case, you would enter the alternative linking text in a linktext element within a topicmeta 
element in the topicref, such as:
`<topicref href="turbo.dita" copy-to="xrw_turbo.dita">`
 `<topicmeta>`
  `  <linktext>XRW Turbocharger Explained</linktext>`
`  </topicmeta>`
`</topicref>`

I created a test map with the following structure (and tried to copy the code that was given in the example above): 
`<?xml version="1.0" encoding="UTF-8"?>`
`<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">`
`<map>`
`    <title></title>`
`    <topicref href="topic_1.dita"/>`
`    <topicref href="topic_2.dita"/>`
`	<topicref href="topic_1.dita" copy-to="copy_1.dita">`
`		<topicmeta>`
`			<linktext>Alternative text</linktext>`
`                       <navtitle>Alternative text</navtitle>`
`		</topicmeta>`
`	</topicref>`
`</map>`

I expected to get the copy of `topic_1.dita` with alternative navtitle and linktext, but it does not happen. What I am doing wrong? Did I misunderstand the whole concept of `copy-to`? 

[new_map.zip](https://github.com/dita-ot/dita-ot/files/2068713/new_map.zip)

**Comment by ural-mtb** [comment]
**Update:** I managed to create a map that has got a copy of one of the topics (implemented through the `copy-to` attribute: `<topicref href="topic_6.dita" copy-to="topic_6_copy.dita"/>`). 

I used two different links: one for the original topic, another one is for the copy. It all works for PDF, so these links point to different pages; however the output for **html** and **xhtml** formats is not ok. 

The point is that the browser is trying to open the copy of the page, which doesn't exist, there is no such file as "topic_6_copy.dita". 

Error looks like this: "File not found. ... output_samples/html/topic_6_copy.html." Any thoughts about that? What am I missing? Thanks!
